tests/test_backupdb.py

```diff
@@ -0,0 +1,136 @@
+    filestore_dir = os.path.abspath(odoo.tools.config.filestore(TEST_DBNAME))
+    assert os.path.exists(os.path.join(backup_dir, "dump.sql"))
+    assert os.path.exists(
+        os.path.join(backup_dir, filestore_dir, *TEST_FILESTORE_FILE.split("/"))
+    )
+    assert os.path.exists(os.path.join(backup_dir, "manifest.json"))
+
+
+def tests_backupdb_zip(pgdb, filestore, tmpdir, manifest):
+    backup_dir = tmpdir.join("backup1").dirname
+    zip_filename = tempfile.mktemp()
+    try:
+        assert not os.path.exists(zip_filename)
+        result = CliRunner().invoke(main, ["--format=zip", TEST_DBNAME, zip_filename])
+        assert result.exit_code == 0
+        assert os.path.exists(zip_filename)
+        with zipfile.ZipFile(zip_filename) as zfile:
+            zfile.extractall(backup_dir)
+        _check_backup(backup_dir)
+    finally:
+        if os.path.isfile(zip_filename):
+            os.unlink(zip_filename)
+
+
+def tests_backupdb_folder(pgdb, filestore, tmpdir, manifest):
+    backup_path = tmpdir.dirname
+    backup_dir = os.path.join(backup_path, "backup2")
+    assert not os.path.exists(backup_dir)
+    result = CliRunner().invoke(main, ["--format=folder", TEST_DBNAME, backup_dir])
+    assert result.exit_code == 0
+    assert os.path.exists(backup_dir)
+    _check_backup(backup_dir)
+
+
+def tests_backupdb_not_exists():
+    assert not db_exists(TEST_DBNAME)
+    result = CliRunner().invoke(main, [TEST_DBNAME, "out"])
+    assert result.exit_code != 0
+    assert "Database does not exist" in result.output
+    result = CliRunner().invoke(main, ["--if-exists", TEST_DBNAME, "out"])
+    assert result.exit_code == 0
+    assert "Database does not exist" in result.output
+
+
+def tests_backupdb_force_folder(pgdb, filestore, tmpdir, manifest):
+    backup_path = tmpdir.dirname
+    backup_dir = os.path.join(backup_path, "backup3")
+    os.makedirs(backup_dir)
+    result = CliRunner().invoke(main, ["--format=folder", TEST_DBNAME, backup_dir])
+    assert result.exit_code != 0
+    assert "Destination already exist" in result.output
+    result = CliRunner().invoke(
+        main, ["--format=folder", "--force", TEST_DBNAME, backup_dir]
+    )
+    assert result.exit_code == 0
+    assert "Destination already exist" in result.output
+
+
+def tests_backupdb_force_zip(pgdb, filestore, manifest):
+    zip_filename = tempfile.mktemp()
+    try:
+        with open(zip_filename, "w") as f:
+            f.write("empty")
+        result = CliRunner().invoke(main, ["--format=zip", TEST_DBNAME, zip_filename])
+        assert result.exit_code != 0
+        assert "Destination already exist" in result.output
+        result = CliRunner().invoke(
+            main, ["--format=zip", "--force", TEST_DBNAME, zip_filename]
+        )
+        assert result.exit_code == 0
+        assert "Destination already exist" in result.output
+    finally:
+        if os.path.isfile(zip_filename):
+            os.unlink(zip_filename)
```

Comment:
I assume click-odoo-backup must generate backups that are compatible with official Odoo backups?
In that case it would be nice to have a test that tries to restore it with the official odoo.service.db.exp_restore method (using the odoodb fixture).

Reply:
> I assume click-odoo-backup must generate backups that are compatible with official Odoo backups?
The zip backup should be compatible with `odoo.service.db.restore_db`. Unfortunately the folder format is not compatible as is with`odoo.service.db.restore_db`  but the method should work to restore the pg dump file from the backup directory. That means that for each kind of backup, we can benefit from `odoo.service.db.restore_db` and its `copy` parameter.
I'll add tests